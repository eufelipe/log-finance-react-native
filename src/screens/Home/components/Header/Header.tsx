import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {SettingsNavigationProp} from 'routes/StacksRoute';
import {useTranslation} from 'react-i18next';

import {
  Container,
  Content,
  Title,
  Description,
  SettingTouchable,
  SettingIcon,
} from './styles';

const Header = (): JSX.Element => {
  const navigation = useNavigation<SettingsNavigationProp>();
  const {t} = useTranslation('home');

  const handleSettings = () =>
    navigation.navigate('SettingsStack', {screen: 'Settings'});

  return (
    <Container>
      <Content>
        <Description>{t('balance')}</Description>
        <Title>R$ -109,90</Title>
      </Content>
      <SettingTouchable onPress={handleSettings}>
        <SettingIcon />
      </SettingTouchable>
    </Container>
  );
};

export default Header;
