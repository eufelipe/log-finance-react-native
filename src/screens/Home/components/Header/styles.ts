import styled from 'styled-components/native';
import {getStatusBarHeight} from 'react-native-iphone-x-helper';
import {Text, TouchableOpacity} from 'styles/layout';

import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Colors from 'styles/colors';

export const Container = styled.View`
  background: ${({theme}) => theme.colors.primary};
  padding: 0px ${({theme}) => theme.spacing.great}px;
  padding-top: ${({theme}) => theme.spacing.tall + getStatusBarHeight()}px;
  justify-content: space-between;
  align-items: center;
  height: 150px;
  flex-direction: row;
`;

export const Content = styled.View``;

export const Title = styled(Text)`
  font-size: ${({theme}) => theme.fontSizes.big};
  font-weight: ${({theme}) => theme.fontWeight.bold};
  font-family: ${({theme}) => theme.fontFamily.semiBold};
  color: ${({theme}) => theme.colors.white};
`;

export const Description = styled(Text)`
  font-size: ${({theme}) => theme.fontSizes.small};
  color: ${({theme}) => theme.colors.white};
`;

export const SettingTouchable = styled(TouchableOpacity).attrs({})`
  padding: 10px;
`;

export const SettingIcon = styled(MaterialIcons).attrs({
  size: 28,
  name: 'settings',
  backgroundColor: 'transparent',
  color: Colors.white,
})`
  padding-right: 0px;
`;
