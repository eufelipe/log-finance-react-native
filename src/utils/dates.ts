import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
