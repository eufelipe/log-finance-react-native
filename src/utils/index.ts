export {default as number} from './number';
export {default as strings} from './strings';
