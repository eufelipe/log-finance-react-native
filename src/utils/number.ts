export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
