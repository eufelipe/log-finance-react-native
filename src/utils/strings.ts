export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
