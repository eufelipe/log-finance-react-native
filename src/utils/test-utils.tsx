/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
