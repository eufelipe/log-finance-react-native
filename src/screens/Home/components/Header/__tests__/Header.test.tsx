/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import {render} from 'utils/test-utils';

import Header from '../Header';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: (key: any) => key}),
}));

describe('Header', () => {
  it('Test match snapshot Header', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<Header {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
