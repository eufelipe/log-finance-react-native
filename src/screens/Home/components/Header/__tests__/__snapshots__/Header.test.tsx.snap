// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test match snapshot Header 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#2B303A",
        "flexDirection": "row",
        "height": 150,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 21.428571428571427,
        "paddingRight": 21.428571428571427,
        "paddingTop": 62.857142857142854,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Poppins-Regular",
            "fontSize": 28,
            "fontWeight": "400",
          },
        ]
      }
    >
      balance
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 60,
            "fontWeight": "700",
          },
        ]
      }
    >
      R$ -109,90
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      backgroundColor="transparent"
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 28,
          },
          Array [
            Object {
              "paddingRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¢¸
    </Text>
  </View>
</View>
`;
